### PR TITLE
Bug/cstm cpc 827 fix reactive owner controller get all owners

### PR DESCRIPTION
**Pull request by Youssefchahboune**
JIRA: https://champlainsaintlambert.atlassian.net/jira/software/c/projects/CPC/boards/14?modal=detail&selectedIssue=CPC-827&sprint=302
## Context:
Worked on the getAllOwners endpoint and the current owners table UI 
## Changes

- getAllOwners endpoint return OwnerResponseDTOs
- Added Tests in customer-service-reactive to test the getAllOwners enpoint
- gave proper naming for all the methods that reteive all owners
- Added Tests in ApiGateway to test the getAllOwners
- Made the current owners table responsive + added filters + changed a little the design 

## Before and After UI (Required for UI-impacting PRs)
Before : 
Screenshot 2023-09-15 181349

After : 
Screenshot 2023-09-19 002803

After : 
image

## Dev notes (Optional)
- 
## Linked pull requests (Optional)
-

### CODE REVIEW

File: api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java [hunk 1]
```diff
@@ -326,8 +326,8 @@ public Mono<Role> addRole(@RequestHeader(AUTHORIZATION) String auth, @RequestBod
      * **/
 
     @GetMapping(value = "owners")
```
Comment (swafit) [inline api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java:328]: should produce TEXT_EVENT_STREAM_VALUE

**Comment by swafit** [comment]
does the controller.js handle the reactive TEXT_EVENT_STREAM_VALUE?

File: api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java (hunk 1 above)
Comment (Youssefchahboune) [reply api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java:328]: What do you mean ?

Comment (Youssefchahboune) [reply api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java:328]: So this should be just fine 
``` $http.get('api/gateway/owners').then(function (resp) {
            self.owners = resp.data;
             console.log(resp)
        }); ```

Comment (Youssefchahboune) [reply api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java:328]: > TEXT_EVENT_STREAM_VALUE

you mean this ``@GetMapping(value = "owners"}, produces = MediaType.TEXT_EVENT_STREAM_VALUE)``

Comment (swafit) [reply api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java:328]: > We are using AngularJS which is a framework that is not inherently reactive like Angular (with RxJS) or other modern frontend frameworks. So basically AngularJS doesn't have built-in support for reactive programming.

It actually is possible to do, I sent you a code snippet and posted it in the general chat on Teams.

**Review by Nmazza1 — APPROVED** [review]
Beasting the front end like usual 👍, I think we might want to consider adding the ability to view the list of pets that may belong to an owner in the future, or a pop up window of sorts when hovered over.

**Review by Saimirz — APPROVED** [review]
Front end looks good. good thing you changed the method names properly making it easier to identify which methods are which.

**Pull request merged**